test/model/internal_link_test.dart

```diff
@@ -0,0 +1,394 @@
+    });
+
+    group('on stream link with wrong name, ID wins', () {
+      checkExpectedStreamNarrow('#narrow/stream/1-nonsense',        1);
+      checkExpectedStreamNarrow('#narrow/stream/1-',                1);
+    });
+
+    group('on malformed stream link: reject', () {
+      checkExpectedStreamNarrow('#narrow/stream/-1',                null);
+      checkExpectedStreamNarrow('#narrow/stream/1nonsense-general', null);
+      checkExpectedStreamNarrow('#narrow/stream/-general',          null);
+    });
+  });
+
+  group('ParseInternalLink with historic links', () {
+    group('for stream with hyphens or even looking like new-style', () {
+      final streams = [
+        eg.stream(streamId: 1, name: 'test-team'),
+        eg.stream(streamId: 2, name: '311'),
+        eg.stream(streamId: 3, name: '311-'),
+        eg.stream(streamId: 4, name: '311-help'),
+        eg.stream(streamId: 5, name: '--help'),
+      ];
+      const testCases = [
+        ('#narrow/stream/test-team/', StreamNarrow(1)),
+        ('#narrow/stream/311/',       StreamNarrow(2)),
+        ('#narrow/stream/311-/',      StreamNarrow(3)),
+        ('#narrow/stream/311-help/',  StreamNarrow(4)),
+        ('#narrow/stream/--help/',    StreamNarrow(5)),
+      ];
+      checkExpectedNarrows(testCases, streams: streams);
+    });
+
+    group('on ambiguous new- or old-style: new wins', () {
+      final streams = [
+        eg.stream(streamId: 1,   name: '311'),
+        eg.stream(streamId: 2,   name: '311-'),
+        eg.stream(streamId: 3,   name: '311-help'),
+        eg.stream(streamId: 311, name: 'collider'),
+      ];
+      const testCases = [
+        ('#narrow/stream/311/',      StreamNarrow(311)),
+        ('#narrow/stream/311-/',     StreamNarrow(311)),
+        ('#narrow/stream/311-help/', StreamNarrow(311)),
+      ];
+      checkExpectedNarrows(testCases, streams: streams);
+    });
+
+    group('on old stream link', () {
+      final streams = [
+        eg.stream(streamId: 1, name: 'check'),
+        eg.stream(streamId: 2, name: 'bot testing'),
+        eg.stream(streamId: 3, name: 'check.API'),
+        eg.stream(streamId: 4, name: 'stream'),
+        eg.stream(streamId: 5, name: 'topic'),
+      ];
+      const testCases = [
+        ('#narrow/stream/check/',         StreamNarrow(1)),
+        ('#narrow/stream/bot.20testing/', StreamNarrow(2)),
+        ('#narrow/stream/check.2EAPI/',   StreamNarrow(3)),
+        ('#narrow/stream/stream/',        StreamNarrow(4)),
+        ('#narrow/stream/topic/',         StreamNarrow(5)),
+
+        ('#narrow/stream/check.API/',     null),
+      ];
+      checkExpectedNarrows(testCases, streams: streams);
+    });
+
+    group('on old stream link, without realm info', () {
+      final stream = eg.stream(name: 'example');
+      final testCases = [
+        ('/#narrow/stream/${stream.name}/', StreamNarrow(stream.streamId)),
+        ('#narrow/stream/${stream.name}/',  StreamNarrow(stream.streamId)),
```

Comment:
The description says "without realm info", but these don't have any less info about the realm than the tests above do.

In the corresponding zulip-mobile tests, the difference is that the other group has absolute URL strings like `https://example.com/#narrow/stream/jest`, while this has relative URL strings. That's already a fairly unnecessary distinction because the test helper there ultimately does `new URL(url, 'https://example.com')` before passing the URL to the code under test; the distinction dates from when the underlying code under test actually took a URL string instead of a `URL` object.

I think these two cases don't add anything not covered in the group above, so we can just drop this one.